### PR TITLE
Improve logs when checking node eligibility to fence a target

### PR DESCRIPTION
**Pull request by kgaillot**

### CODE REVIEW

File: daemons/fenced/fenced_commands.c [hunk 1]
```diff
@@ -2081,42 +2081,43 @@ localhost_is_eligible(const stonith_device_t *device, const char *action,
...
          * we'll report that with the results. We never allow suicide for
          * remapped "on" operations because the host is off at that point.
          */
-        if (!localhost_is_eligible(dev, "reboot", host, search->allow_suicide)
-            && !localhost_is_eligible(dev, "off", host, search->allow_suicide)
-            && !localhost_is_eligible(dev, "on", host, FALSE)) {
+        if (!localhost_is_eligible(dev, "reboot", target, search->allow_suicide)
+            && !localhost_is_eligible(dev, "off", target, search->allow_suicide)
+            && !localhost_is_eligible(dev, "on", target, FALSE)) {
             goto search_report_results;
```
Comment (wenningerk) [inline daemons/fenced/fenced_commands.c:2125]: when remapping we might remap "reboot" to a sole "off" but shouldn't unavailability of both "reboot" & "off" already be a hard criterion that it won't fit?

File: daemons/fenced/fenced_commands.c [hunk 2]
```diff
@@ -2081,42 +2081,43 @@ localhost_is_eligible(const stonith_device_t *device, const char *action,
...
+    target = search->host;
+    if (target == NULL) {
         can = TRUE;
         goto search_report_results;
     }
 
     /* Short-circuit query if this host is not allowed to perform the action */
-    if (pcmk__str_eq(search->action, "reboot", pcmk__str_casei)) {
+    if (pcmk__str_eq(action, "reboot", pcmk__str_casei)) {
         /* A "reboot" *might* get remapped to "off" then "on", so short-circuit
```
Comment (wenningerk) [inline daemons/fenced/fenced_commands.c:2116]: isn't that one of the case-in-sensitive comparisons we're removing on the way ...

File: daemons/fenced/fenced_commands.c [hunk 3]
```diff
@@ -2104,6 +2104,13 @@ can_fence_host_with_device(stonith_device_t *dev,
         goto search_report_results;
     }
 
+    // Short-circuit if device does not support action
+    if (pcmk__str_eq(action, "on", pcmk__str_none)
+        && !pcmk_is_set(dev->flags, st_device_supports_on)) {
+        check_type = "Agent does not support 'on'";
+        goto search_report_results;
+    }
+
```
Comment (wenningerk) [inline daemons/fenced/fenced_commands.c:2113]: We could combine that to
/* Short-circuit query if device doesn't allow action or
    this host is not allowed to perform the action */
removing a compare.
If it improves/simplifies readability probably depends on personal preference.

**Comment by wenningerk** [comment]
Looking good to me.

File: daemons/fenced/fenced_commands.c (hunk 1 above)
Comment (kgaillot) [reply daemons/fenced/fenced_commands.c:2125]: I'm not 100% sure, but I believe it's possible for different devices (or at least nodes) to be chosen for the "off" and "on" portions, so it's still worth telling the initiator if this node can only do one of them

File: daemons/fenced/fenced_commands.c (hunk 3 above)
Comment (kgaillot) [reply daemons/fenced/fenced_commands.c:2113]: makes sense

File: daemons/fenced/fenced_commands.c (hunk 1 above)
Comment (wenningerk) [reply daemons/fenced/fenced_commands.c:2125]: Hadn't checked for the actual flow of things and can't recall from memory.

**Comment by kgaillot** [comment]
Added two commits on top to address review

**Comment by kgaillot** [comment]
Updated top commit to improve readability

**Pull request merged**